src/transaction/transaction.rs

```diff
@@ -11,27 +12,12 @@ use std::ops::RangeBounds;
 /// particular timestamp obtained from the placement driver.
 ///
 /// Once a transaction is commited, a new commit timestamp is obtained from the placement driver.
+#[derive(new)]
 pub struct Transaction {
     snapshot: Snapshot,
 }
 
 impl Transaction {
-    /// Create a new transaction operating on the given snapshot.
```

Comment:
Seems this documentation will be gone now, could we keep it?

Reply:
Sure, we can move it to the struct definition.